internal/ui/etui/ephemeral_tui.go

```diff
@@ -73,16 +75,25 @@ func OutputToEphemeralTUI(workerErrs <-chan error, subscription *partybus.Subscr
 	ctx := context.Background()
 	grypeUIHandler := grypeUI.NewHandler()
 
-eventLoop:
+	var errResult error
 	for {
 		select {
-		case err := <-workerErrs:
+		case err, ok := <-workerErrs:
 			if err != nil {
+				if errors.Is(err, grypeerr.ErrAboveSeverityThreshold) {
+					errResult = err
+					continue
+				}
 				return err
 			}
+			if !ok {
+				// worker completed
+				workerErrs = nil
+			}
 		case e, ok := <-events:
```

Comment:
GoLand underlined `events` with this warning: "Receive may block because of 'nil' channel". May be worth checking for `nil` ahead of time.

Reply:
per our conversation, this section probably needs more love than this PR is willing to give